src/PhoneNumberInput/PhoneNumberInput.jsx

```diff
@@ -55,6 +56,12 @@ const PhoneNumberInput = forwardRef((props, ref) => {
   const [country, setCountry] = useState(statesEu[0])
   const [dropdownOpen, setDropdownOpen] = useState(false)
 
+  const [cursorPosition, setCursorPosition] = useState(0)
```

Comment:
Drop this state, it's not relevant. You can always read the cursor position, you don't need to keep an out-of-sync value in memory

Reply:
This value is needed because the modification of the cursor position happens async, between the trimming and the DOM input processing. This ensures that I have the correct selection position before the field value gets changed.